### PR TITLE
Split cloud and robot master into app-rollout and chart-assignment controllers

### PR DESCRIPTION
**Pull request by oliver-goetz**
The functionality of cloud-master and robot-master is now split into app-rollout-controller and chart-assignment-controller apps.

app-rollout-controller is supposed to run in the cloud cluster only.
chart-assignment-controller runs in both cluster types, cloud and robot. It has a slightly different configuration for cloud and robot deployment regarding webhook configuration and stackdriver log upload.

### CODE REVIEW

File: src/app_charts/base/cloud/app-management.yaml [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+          value: {{ .Values.project }}
+        ports:
+        - name: webhook
+          containerPort: 9876
+        volumeMounts:
+        - mountPath: /home
+          name: home
+        - mountPath: /tls
+          name: tls
+      initContainers:
```
Comment (ensonic) [inline src/app_charts/base/cloud/app-management.yaml:41]: we probably only need this for the chart assignment controller. The approll-out is only creating the chart assignments.

File: src/go/cmd/chart-assignment-controller/main.go [hunk 2]
```diff
@@ -111,12 +120,20 @@ func setupAppV2(cfg *rest.Config, cluster string) error {
...
-		webhook := chartassignment.NewValidationWebhookForEdgeCluster(mgr, cluster)
-		srv.Register("/chartassignment/validate", webhook)
+			srv.Register("/chartassignment/validate", chartassignment.NewValidationWebhook(mgr))
+		} else {
+			srv := mgr.GetWebhookServer()
+			srv.CertDir = *certDir
+
+			webhook := chartassignment.NewValidationWebhookForEdgeCluster(mgr, cluster)
+			srv.Register("/chartassignment/validate", webhook)
+		}
```
Comment (ensonic) [inline src/go/cmd/chart-assignment-controller/main.go:136]: this block has some code duplication: Maybe:
```go
var webhook *admission.Webhook
if *cloudCluster {
  webhook = chartassignment.NewValidationWebhook(mgr)
} else {
  webhook = chartassignment.NewValidationWebhookForEdgeCluster(mgr, cluster)
}
srv := mgr.GetWebhookServer()
srv.CertDir = *certDir
srv.Register("/chartassignment/validate", webhook)
```

**Comment by ensonic** [comment]
Awesome. This is looking great. I'll pull this soon into our CI to run the tests.

File: src/app_charts/base/cloud/app-management.yaml [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Certificate
+metadata:
+  name: chart-assignment-controller
+spec:
+  secretName: chart-assignment-controller-tls
+  commonName: chart-assignment-controller.{{ .Release.Namespace }}.svc
+  issuerRef:
+    kind: ClusterIssuer
+    name: cluster-authority
```
Comment (ensonic) [inline src/app_charts/base/cloud/app-management.yaml:172]: We might only want to do this in the cloud (where cert-manager creates the cert), for the robot counterpart, we create it in setup-robot

Comment (ensonic) [reply src/app_charts/base/cloud/app-management.yaml:172]: Ignore me. This is cloud only :)

File: src/app_charts/base/cloud/app-management.yaml (hunk 1 above)
Comment (oliver-goetz) [reply src/app_charts/base/cloud/app-management.yaml:41]: Oh yes, that's true. We could remove the helm binary from app-rollout-controller too. I'll test that.

Comment (oliver-goetz) [reply src/app_charts/base/cloud/app-management.yaml]: This is working. I removed the helm binary, initContainer and /home volume mount from app-rollout-controller.

File: src/go/cmd/chart-assignment-controller/main.go (hunk 2 above)
Comment (oliver-goetz) [reply src/go/cmd/chart-assignment-controller/main.go:131]: done

**Pull request closed without merging**